spec/watch_spec.rb

```diff
@@ -12,207 +20,368 @@
 
     it "yields create_initial, modify, timeout then modify, timeout file events" do
-      write_lines_1_and_2_proc.call
-      write_lines_3_and_4_proc.call # delayed async call
-      subject.watch(File.join(directory, "*"))
-      quit_proc.call
       subscribe_proc.call
       expect(results).to eq([[:create_initial, file_path], [:modify, file_path], [:timeout, file_path], [:modify, file_path], [:timeout, file_path]])
     end
   end
 
   context "when ignore older expiry is enabled and all files are already expired" do
-    let(:quit_sleep) { 3 }
-    let(:stat_interval) { 0.2 }
-
     before do
-      File.open(file_path, "wb") { |file|  file.write("line1\nline2\n") }
       subject.ignore_older = 1
+
+      RSpec::Sequencing
+        .run("create file") do
+          File.open(file_path, "wb") { |file|  file.write("line1\nline2\n") }
+        end
+        .then_after(2, "start watching after file ages more than ignore_older") do
+          subject.watch(watch_dir)
+        end
+        .then_after(1, "quit after allowing time to close the file") do
+          subject.quit
+        end
     end
 
-    it "yields only create_initial file event" do
-      sleep 2
-      subject.watch(File.join(directory, "*"))
-      quit_proc.call
+    it "yields no file events" do
       subscribe_proc.call
-      expect(results).to eq([[:create_initial, file_path]])
+      expect(results).to eq([])
     end
   end
 
   context "when ignore_older is less than close_older and all files are not expired" do
-    let(:quit_sleep) { 3 }
-    let(:stat_interval) { 0.2 }
-
     before do
-      File.open(file_path, "wb") { |file|  file.write("line1\nline2\n") }
       subject.ignore_older = 1
       subject.close_older = 2
+
+      RSpec::Sequencing
+        .run("create file") do
+          File.open(file_path, "wb") { |file|  file.write("line1\nline2\n") }
+        end
+        .then("start watching before file age reaches ignore_older") do
+          subject.watch(watch_dir)
+        end
+        .then_after(3.1, "quit after allowing time to close the file") do
+          subject.quit
+        end
     end
 
     it "yields create_initial, modify, timeout file events" do
-      subject.watch(File.join(directory, "*"))
-      quit_proc.call
       subscribe_proc.call
       expect(results).to eq([[:create_initial, file_path], [:modify, file_path], [:timeout, file_path]])
     end
   end
 
   context "when ignore_older is less than close_older and all files are expired" do
-    let(:quit_sleep) { 3 }
-    let(:stat_interval) { 0.2 }
-
     before do
-      File.open(file_path, "wb") { |file|  file.write("line1\nline2\n") }
       subject.ignore_older = 1
       subject.close_older = 2
+
+      RSpec::Sequencing
+        .run("create file") do
+          File.open(file_path, "wb") { |file|  file.write("line1\nline2\n") }
+        end
+        .then_after(1.95, "start watching after file ages more than ignore_older") do
+          subject.watch(watch_dir)
```

Comment:
We cause `File::utime(...)` to force the mtime of the file to be whatever we wish. It'll save us some sleeps.